### PR TITLE
PHP 8 Action workflows

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
Enables PHP 8 in the relevant CI workflows. This PR will require #3938 and a rebase before being close to passing.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: .github/workflows/test-phpstan.yml [hunk 1]
```diff
@@ -20,16 +20,20 @@ on:
 
 jobs:
   build:
-    name: Analyze code (PHPStan)
+    name: PHP ${{ matrix.php-versions }} Analysis
```
Comment (paulbalandan) [inline .github/workflows/test-phpstan.yml:23]:
```suggestion
    name: PHP ${{ matrix.php-versions }} Static Analysis
```

Comment (MGatner) [reply .github/workflows/test-phpstan.yml:23]: 👍 I'm not ready to kick of CI for this again but I will implement this when I am.

**Comment by MGatner** [comment]
This is awaiting #3940 and another rebase of 4.1 to resolve the Autoloader merge.

**Comment by MGatner** [comment]
```
✗ imagick Could not install imagick on PHP 8.0.0
✗ sqlsrv Could not install sqlsrv on PHP 8.0.0
```
Supposedly `imagick` should be working for PHP 8.0 so I'm not sure why it is not installing? https://github.com/shivammathur/setup-php/issues/320

**Comment by MGatner** [comment]
`imagick` is not on the supported list for `8.0` (https://github.com/shivammathur/setup-php/wiki/Php-extensions-on-ubuntu-20.04#php-80) but it also isn't on the `7.4` list so maybe that just has not been updated?

**Comment by MGatner** [comment]
Ref. https://github.com/shivammathur/setup-php/issues/353

**Comment by MGatner** [comment]
Since 4.0.5 is now officially going to support PHP 8 this should be redirected to `develop` and resolved there.

**Comment by MGatner** [comment]
Starting afresh.

**Pull request closed without merging**